projects/container_bench/toolbox/container_bench_create_container_benchmark/tasks/main.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Validate required variables are defined
+  assert:
+    that:
+      - container_bench_create_container_benchmark_exec_time_path is defined
+      - container_bench_create_container_benchmark_binary_path is defined
+      - container_bench_create_container_benchmark_rootfull is defined
+      - container_bench_create_container_benchmark_additional_args is defined
```

Comment:
I'm not sure to understand why you assert that? while this already ensures that it will be there:
```
    def create_container_benchmark(
            self,
            exec_time_path,
            binary_path,
            rootfull=False,
            additional_args="",
    ):
    ```
is it to run without the Python wrapper?

Reply:
```
- name: Map long variable names to short aliases for readability
  set_fact:
    binary_path: "{{ vars[benchmark_role_name + '_binary_path'] }}"
    additional_args: "{{ vars[benchmark_role_name + '_additional_args'] }}"
    exec_time_path: "{{ vars[benchmark_role_name + '_exec_time_path'] }}"
    rootfull: "{{ vars[benchmark_role_name + '_rootfull'] }}"
```
got it.
One thing you may do to avoid that is to pass a dict from Python to ansible
something like
```
run_toolbox(exec_props=dict(path=..., args=...., path=...))
```
then in Ansible
```
set_facts:
  exec_props: my_role_exec_props # this one needs to change for each role

set_facts:
  path: exec_props["path"]
  args: exec_props["args"]
```

(not sure which version is cleaner though ^^)